espnet/nets/pytorch_backend/transformer/subsampling.py

```diff
@@ -20,9 +20,59 @@ class Conv2dSubsampling(torch.nn.Module):
 
     """
 
-    def __init__(self, idim, odim, dropout_rate):
+    def __init__(self, idim, odim, dropout_rate, pos_enc=None):
         """Construct an Conv2dSubsampling object."""
         super(Conv2dSubsampling, self).__init__()
+        if pos_enc:
+            self.pos_enc = pos_enc
+        else:
+            self.pos_enc = PositionalEncoding(odim, dropout_rate)
+
+        self.conv = torch.nn.Sequential(
+            torch.nn.Conv2d(1, odim, 3, 2),
+            torch.nn.ReLU(),
+            torch.nn.Conv2d(odim, odim, 3, 2),
+            torch.nn.ReLU()
+        )
+        self.out = torch.nn.Sequential(
+            torch.nn.Linear(odim * (((idim - 1) // 2 - 1) // 2), odim),
+            self.pos_enc
+        )
+
+    def forward(self, x, x_mask):
+        """Subsample x.
+
+        :param torch.Tensor x: input tensor
+        :param torch.Tensor x_mask: input mask
+        :return: subsampled x and mask
+        :rtype Tuple[torch.Tensor, torch.Tensor]
+        """
+        x = x.unsqueeze(1)  # (b, c, t, f)
+        x = self.conv(x)
+        b, c, t, f = x.size()
+        x = self.out(x.transpose(1, 2).contiguous().view(b, t, c * f))
+        if x_mask is None:
+            return x, None
+        return x, x_mask[:, :, :-2:2][:, :, :-2:2]
+
+class Conv2dSubsamplingCustomPosEnc(torch.nn.Module):
+    """Convolutional 2D subsampling (to 1/4 length) w/ custom positional encoding.
+       This is used in VTN encoder.
+
+    :param int idim: input dim
+    :param int odim: output dim
+    :param flaot dropout_rate: dropout rate
+
+    """
+
+    def __init__(self, idim, odim, dropout_rate, pos_enc=None):
+        """Construct an Conv2dSubsampling object."""
+        super(Conv2dSubsamplingCustomPosEnc, self).__init__()
```

Comment:
Is it needed to separate module?
It can be implemented with options.
I think if you set default value and keep the same behavior when the option is default, it is OK.